src/node_process_methods.cc

```diff
@@ -242,6 +246,28 @@ static void Uptime(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(result);
 }
 
+static Maybe<void> GetResourceInfo(Environment* env, AsyncWrap* w,
+                                   std::vector<Local<Value>>* resources) {
+  Local<Object> resource = Object::New(env->isolate());
+
+  if (resource->Set(env->context(),
+                    OneByteString(env->isolate(), "type"),
+                    OneByteString(env->isolate(), w->MemoryInfoName().c_str()))
+                        .IsNothing() ||
+      resource->Set(env->context(),
+                    OneByteString(env->isolate(), "asyncId"),
+                    Number::New(env->isolate(), w->get_async_id()))
+                        .IsNothing() ||
+      resource->Set(env->context(),
+                    OneByteString(env->isolate(), "triggerAsyncId"),
+                    Number::New(env->isolate(), w->get_trigger_async_id()))
+                        .IsNothing()) return Nothing<void>();
+
+  resources->emplace_back(resource);
```

Comment:
Personally I would leave this in the other functions and have this function only build and return a Maybe of the Local<Object>. Eliminates the logical connection to putting the element in an array which makes this method a little more flexible. Just a suggestion though, I won't block on that, it LGTM as-is. 😄

Reply:
@Qard sure, done!